packages/devtools_server/lib/src/file_system.dart

```diff
@@ -43,8 +43,13 @@ class LocalFileSystem {
 
   /// Returns a DevTools file from the given path.
   ///
-  /// Only files withing ~/.devtools/ can be accessed.
+  /// Only files within ~/.devtools/ can be accessed.
   static File devToolsFileFromPath(String pathFromDevToolsDir) {
+    if (pathFromDevToolsDir.contains('../')) {
```

Comment:
more robust would be to use 
https://api.dart.dev/stable/2.10.4/dart-io/FileSystemEntity/resolveSymbolicLinks.html
to resolve all . and ../
you could also fail if there is any `..` in the path.

Reply:
returned null if .. is present